webapp-django/crashstats/crashstats/jinja2/crashstats/report_index.html

```diff
@@ -321,13 +321,24 @@ <h2>{{ report.product }} {{ report.version }} Crash Report [@ {{ report.signatur
                                 <th scope="row">Crash Address</th>
                                 <td>{{ report.address }}</td>
                             </tr>
-                            {% if raw.JavaStackTrace %}
-                            <tr title="{{ fields_desc['processed_crash.java_stack_trace'] }}">
-                                <th scope="row">Java Stack Trace</th>
-                                <td>
-                                    <pre>{{ raw.JavaStackTrace }}</pre>
-                                </td>
-                            </tr>
+                            {% if request.user.has_perm('crashstats.view_pii') or your_crash %}
+                                {% if report.java_stack_trace_full %}
+                                <tr title="{{ fields_desc['processed_crash.java_stack_trace_full'] }}">
+                                    <th scope="row">Java Stack Trace (Full)</th>
+                                    <td>
+                                        <pre>{{ report.java_stack_trace_full }}</pre>
+                                    </td>
+                                </tr>
+                                {% endif %}
+                            {% else %}
+                                {% if report.java_stack_trace %}
+                                <tr title="{{ fields_desc['processed_crash.java_stack_trace'] }}">
+                                    <th scope="row">Java Stack Trace (Redacted)</th>
+                                    <td>
+                                        <pre>{{ report.java_stack_trace }}</pre>
+                                    </td>
+                                    </tr>
+                                {% endif %}
```

Comment:
If the user is logged in and has `view_pii`, then the user will see `java_stack_trace_full`. Otherwise the user will see `java_stack_trace`. I think that works nicely.